crates/matrix-sdk-ui/src/timeline/event_item/content.rs

```diff
@@ -831,6 +865,72 @@ impl AnyOtherFullStateEventContent {
             Self::_Custom { event_type } => event_type.as_str().into(),
         }
     }
+
+    fn redact(&self, room_version: &RoomVersionId) -> Self {
+        match self {
+            Self::PolicyRuleRoom(c) => Self::PolicyRuleRoom(FullStateEventContent::Redacted(
+                c.clone().redact(room_version),
```

Comment:
Here and below, are the `clone()` necessary? Seems like the `redact()` function always take `&self` and return a new event anyways.

Reply:
This is `ruma_common::events::RedactContent::redact`, which takes `self`.